universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java

```diff
@@ -164,4 +161,35 @@ private Result withMessage(Result result, String message) {
     }
     return result;
   }
+
+  private QuestionForm createQuestionForm(Request request, String type)
+      throws InvalidQuestionTypeException {
+    QuestionType questionType;
+    try {
+      questionType = QuestionType.of(type);
+    } catch (InvalidQuestionTypeException e) {
+      throw new InvalidQuestionTypeException(
+          String.format(
+              "unrecognized question type: '%s', accepted values include: %s",
+              type.toUpperCase(), Arrays.toString(QuestionType.values())));
+    }
+
+    switch (questionType) {
+      case TEXT:
+        {
+          Form<TextQuestionForm> form = formFactory.form(TextQuestionForm.class);
+          // Note: We add discardingErrors() to get rid of unhelpful errors that are thrown on empty
+          // number inputs. If we find there is a better solution out there, we should update to use
+          // that. But since we don't rely on the spring form data binder system for validation, we
+          // can "safely" ignore these errors.
+          return form.bindFromRequest(request).discardingErrors().get();
+        }
+      default:
+        {
+          // TODO(natsid): Once QuestionForm is abstract, the default case should throw.
```

Comment:
question: do we have an issue for making `QuestionForm` abstract? If so, could we link it here?

Reply:
We do not have a specific issue for that. I was just going to do it as part of #589. Can link it when I get back this afternoon.

Reply:
Actually I just linked it now.